### PR TITLE
test: When executing js command that doesn't return any value, don't …

### PR DESCRIPTION
**Pull request by marusak**
…expect it

Also sosreport tests are broken, as when it tries to extract the archive, it cannot extract `/procs` due to lack of permissions. Wonder how this differ between chrome and firefox...

### CODE REVIEW

**Comment by martinpitt** [comment]
This seems to work in general, but breaks one test:

    TypeError: inject_js() takes 2 positional arguments but 3 were given

**Comment by martinpitt** [comment]
[testCentos7](https://logs.cockpit-project.org/logs/pull-14199-20200605-030853-06fbc7e3-fedora-31/log.html#22-2) is now broken with chromium: "Object reference chain is too long". It apparently doesn't like wrapping an extra dictionary around the returned promise?

File: test/verify/check-multi-os [hunk 1]
```diff
@@ -50,7 +50,7 @@ class TestMultiOS(MachineCase):
     }
 
     def check_spawn(self, b, address):
-        # Firefox cannot do `cockpit.spawn` as it returns promise
+        # HACK: Firefox cannot do `cockpit.spawn` as it returns promise
```
Comment (marusak) [inline test/verify/check-multi-os:53]: This message does not seem to be fully correct.
Firefox did not support promises, but it seems it does now (\o/). So this one technically should work. I'll take a look if we can really drop it in different PR.

File: test/verify/check-multi-os [hunk 2]
```diff
@@ -59,9 +59,12 @@ class TestMultiOS(MachineCase):
         self.assertEqual(result, "hi\n")
 
     def check_dbus(self, b, address):
+        # HACK: Firefox cannot do `cockpit.dbus.proxy.call` as it returns promise
```
Comment (marusak) [inline test/verify/check-multi-os:62]: This message seems wrong then as well. But if tests pass, lets not block on that.

**Comment by marusak** [comment]
So now firefox can wait for promise to resolve, but cannot get value from them. I also did some cleanup in firefox cdp driver, lets see how it stands in CI.

[this](https://chromedevtools.github.io/devtools-protocol/tot/Runtime/#method-awaitPromise) might help, but I need to check that out firstly.

**Comment by marusak** [comment]
Sosreport and [testPages](https://logs.cockpit-project.org/logs/pull-14199-20200605-055551-8147163e-fedora-31-firefox/log.html#1-2) which points to two different issues. One being that it fails somehow somewhere and then fails to do proper cleanup of the VM.

**Pull request merged**